src/main.c

```diff
@@ -173,6 +217,63 @@ void WriteInterfaces()
     free(InterfaceArray);
 }
 
+inline int
+CombineMetadataWithPacket(
+    _In_ HANDLE File,
+    _In_ PCHAR FragBuf,
+    _In_ unsigned long FragLength,
+    _In_ long InterfaceId,
+    _In_ long IsSend,
+    _In_ long TimeStampHigh, // usec (unless if_tsresol is used)
+    _In_ long TimeStampLow,
+    _In_ PDOT11_EXTSTA_RECV_CONTEXT Metadata,
+    _In_ unsigned long ProcessId
+)
+{
+
+    char Comment[MAX_PACKET_SIZE] = { 0 };
+    size_t commentlength = 0;
+
+    HRESULT Err = StringCchPrintfA((char*)&Comment, MAX_PACKET_SIZE, "Packet Metadata: ReceiveFlags:0x%x, PhyType:%s, CenterCh:%u, NumMPDUsReceived:%u, RSSI:%d, DataRate:%u, PID=%d",
+        Metadata->uReceiveFlags,
+        DOT11_PHY_TYPE_NAMES[Metadata->uPhyId],
+        Metadata->uChCenterFrequency,
+        Metadata->usNumberOfMPDUsReceived,
+        Metadata->lRSSI,
+        Metadata->ucDataRate,
+        ProcessId);
+
+    if (FAILED(Err))
+    {
+        printf("Failed converting NdisMetadata to string with error: %u\n", Err);
+    }
+    else
+    {
+        Err = StringCchLengthA((char*)&Comment, MAX_PACKET_SIZE, &commentlength);
+
+        if (FAILED(Err))
```

Comment:
Please use existing style (i.e. check Err != NO_ERROR rather than using FAILED macro) unless there is a strong reason not to. And you decide to use FAILED, change all error checks to use FAILED macro rather than just the new code. Don't introduce an inconsistency.

Reply:
Hmmm, I was following the new code / standard that was added with one of the latest commits. Originally I was following the existing pattern, but changed my code only after seeing the new macros go in.

https://github.com/microsoft/etl2pcapng/commit/d5de9d1848d21cf8975b832a9556169dd8e24344

I can work on updating them all though, sure.

Reply:
Oh yikes, I missed it in that change then. Thanks for updating the types!